docs/en/advanced_guides/detection_transformer.md

```diff
@@ -0,0 +1,230 @@
+
+```Python
+class DetrTransformerEncoder(BaseModule):
+    def __init__(self,
+                 num_layers: int,
+                 layer_cfg: ConfigType,
+                 init_cfg: OptConfigType = None) -> None:
+        super().__init__(init_cfg=init_cfg)
+        self.num_layers = num_layers
+        self.layer_cfg = layer_cfg
+        self._init_layers()
+
+    def _init_layers(self) -> None:
+        self.layers = ModuleList([
+            DetrTransformerEncoderLayer(**self.layer_cfg)
+            for _ in range(self.num_layers)])
+        self.embed_dims = self.layers[0].embed_dims
+
+    def forward(self, query: Tensor, query_pos: Tensor,
+                key_padding_mask: Tensor, **kwargs) -> Tensor:
+        for layer in self.layers:
+            query = layer(query, query_pos, key_padding_mask, **kwargs)
+        return query
+```
+
+Additionally, the `XHead` becomes lightweight, because the Transformer components are moved to detector, which makes the head focus on getting detection results and calculating losses.
+
+#### 3. Code with better readability and usability
+
+The code of all DETR-related modules are refactored to enhance the readability and usability.
+
+The refactored modules have reasonable designs and a uniform implementation style, which makes it easy to read the code of supported DETRs or implement a custom DETR. Users can refer to [Customize a DETR](<>) to implement a new DETR.
+
+The overused registration mechanism for building modules are replaced with direct initialization, which benefits code reading and jumping.
+
+```python
+# Original implementation
+self.encoder = build_transformer_layer_sequence(encoder)
+self.decoder = build_transformer_layer_sequence(decoder)
+# Refactored implementation
+self.encoder = DetrTransformerEncoder(**self.encoder)
+self.decoder = DetrTransformerDecoder(**self.decoder)
+# Show the module class directly, support code jumping.
+```
+
+At last, the new modules come with more detailed doc-strings and comments.
+
+#### 4. More SOTA DETRs
+
+In the latest MMDetection, the supported DETRs have been refactored. Besides, implementation and weight of several SOTA DETRs have been available.
+
+Supported DETRs:
+
+- [x] [DETR](https://arxiv.org/abs/2005.12872) (ECCV'2020)
+- [x] [Deformable-DETR](https://arxiv.org/abs/2010.04159v4) (ICLR'2021)
+- [x] [Conditional-DETR](https://arxiv.org/abs/2108.06152v2) (ICCV'2021)
+- [x] [DAB-DETR](https://arxiv.org/abs/2201.12329) (ICLR'2022)
+- [x] [DINO](https://arxiv.org/abs/2203.03605) (ArXiv'2022)
+
+### Basic conceptions
+
+#### Image feature and sequence feature
+
+Transformer is the prevalent architecture for nature language processing, which usually process sequence data. There are usually three dimensions of sequence data: batch size `B`, number of tokens / length of sequence `N`, embedding dimension / number of channels `C`.
```

Comment:
Transformer is a prevalent architecture for nature language processing, which usually processes sequence data. Typically, sequence data has three dimensions: batch size `B`, number of tokens / length of sequence `N`, embedding dimension / number of channels `C`.

Reply:
> Transformer is a prevalent architecture for nature language processing, which usually processes sequence data. Typically, sequence data has three dimensions: batch size `B`, number of tokens / length of sequence `N`, embedding dimension / number of channels `C`.

Thanks very much for your advice!